File: java/lib/SnobotSim/ModuleWrapper/DigitalSourceWrapper.cpp
/*
 * DigitalSourceWrapper.cpp
 *
 *  Created on: May 3, 2017
 *      Author: PJ
 */

#include "SnobotSim/ModuleWrapper/DigitalSourceWrapper.h"

DigitalSourceWrapper::DigitalSourceWrapper(int aPort) :
        AModuleWrapper("Digital Source" + std::to_string(aPort)), mState(false)
{

}

DigitalSourceWrapper::~DigitalSourceWrapper()
{

}


bool DigitalSourceWrapper::Get()
{
    return mState;
}

void DigitalSourceWrapper::Set(bool aState)
{
    mState = aState;
}

File: snobot_sim_java/src/main/java/com/snobot/simulator/module_wrapper/DigitalSourceWrapper.java
package com.snobot.simulator.module_wrapper;

public class DigitalSourceWrapper extends ASensorWrapper
{
    private final StateSetterHelper mSetterHelper;
    private boolean mState;

    public static interface StateSetterHelper
    {
        public void setState(boolean aState);
    }

    public DigitalSourceWrapper(int aIndex, StateSetterHelper aSetterHelper)
    {
        super("Digital Source" + aIndex);

        mSetterHelper = aSetterHelper;
        mState = true;
    }

    public boolean get()
    {
        return mState;
    }

    public void set(boolean aState)
    {
        boolean isUpdate = mState != aState;
        mState = aState;
        if (isUpdate)
        {
            mSetterHelper.setState(mState);
        }

    }
}
